tools/travis-scripts/run-script.sh

```diff
@@ -37,6 +37,16 @@ function build_linux()
     cmake --build .
 }
 
+function build_linux_clang_tidy()
+{
+    echo "Building clang tidy test on Linux ..."
+    cd $COCOS2DX_ROOT/build
+    mkdir -p clang-tidy-build
+    cd clang-tidy-build
+    cmake ../.. -DCMAKE_EXPORT_COMPILE_COMMANDS=on
+    python /usr/local/clang-7.0.0/share/clang/run-clang-tidy.py
```

Comment:
```
cmake ../.. -DCMAKE_EXPORT_COMPILE_COMMANDS=on
python /usr/local/clang-7.0.0/share/clang/run-clang-tidy.py
```

Don't quite understand these codes. And the python script path is hard coded, it may be a problem when CI changes its environment.

Reply:
I `apt search clang-tidy` in travis ci and found the most recent version is 6. But the `clang-tidy` being used is 7. My guess is that `clang-tidy` is not installed by package manager. I am not sure how to find `run-clang-tidy.py`.

Or we can just copy one and store it in the repo, if you agree.

Reply:
@minggo 
`cmake ../.. -DCMAKE_EXPORT_COMPILE_COMMANDS=on`
CMAKE_EXPORT_COMPILE_COMMANDS is used to generate compilation database. This is the easiest way to use clang-tidy with CMake 
https://clang.llvm.org/docs/JSONCompilationDatabase.html

About hard-coded path 
`python /usr/local/clang-7.0.0/share/clang/run-clang-tidy.py`
It can be changed to e.g.:
"python $(dirname $(dirname $(readlink -f \`which clang-tidy\`)))/share/clang/run-clang-tidy.py"

Reply:
@crazyhappygame Great. It works on Fedora Linux too. I didn't know clang-tidy shares top-level directory with its python script.

Reply:
@crazyhappygame thanks for the explanation.